src/geometry/BoltSceneNode.cxx

```diff
@@ -159,33 +159,23 @@ void            BoltSceneNode::addLight(
 
 void            BoltSceneNode::notifyStyleChange()
 {
-    texturing = BZDBCache::texture && BZDBCache::blend;
+    texturing = BZDBCache::texture;
     OpenGLGStateBuilder builder(gstate);
     builder.enableTexture(texturing);
-    if (BZDBCache::blend)
-    {
-        const int shotLength = (int)(BZDBCache::shotLength * 3.0f);
-        if (shotLength > 0 && !drawFlares)
-            builder.setBlending(GL_SRC_ALPHA, GL_ONE);
-        else
-            builder.setBlending(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
-        builder.setStipple(1.0f);
-        builder.setAlphaFunc();
-        if ((RENDERER.useQuality() >= 3) && drawFlares)
-        {
-            builder.setShading(GL_SMOOTH);
-            builder.enableMaterial(false);
-        }
-        else
-            builder.setShading(texturing ? GL_FLAT : GL_SMOOTH);
-    }
+    const int shotLength = (int)(BZDBCache::shotLength * 3.0f);
```

Comment:
C-style casts are evil. Either use a C++ cast or a "function-style" cast (anonymous temporary:
```c++
const int shotLength = static_cast<int>(BZDBCache::shotLength * 3.0);
const int shotLength = int( BZDBCache::shotLength * 3.0 );
```
(although I don't know why we are multiplying by a float anyway)
(and magic numbers like "3.0" are maintenance burdens)
(and I realize you just removed the `if` but any time you touch a line of code, it's nice to leave it better)

Reply:
I don't know why we multiply it either.
Removing

Reply:
I wasn’t suggesting the multiplication was bad, just that multiplying by `3.0f` vs `3` was silly, especially given the conversion to `int`

Reply:
I re-added it

Reply:
> I wasn’t suggesting the multiplication was bad, just that multiplying by `3.0f` vs `3` was silly, especially given the conversion to `int`

I just wanted to point out that the result is not necessarily the same (if the value has a fractional part).

Reply:
Not to quibble but it is the same. `1.1 * 3.0 == 1.1 * 3`. Integer and floating point numeric operations are well defined and well behaved

Reply:
Okay I see what you were saying. I thought you were suggesting to do the multiplication after the cast.